doc/build_docs.py

```diff
@@ -38,6 +38,34 @@
 _DOCTREE_TMP = _TMP / "doctree"
 
 
+class ShardingStrategy:
+    def __init__(self, spec: str) -> None:
+        self.spec = spec
+        if spec == "no":
+            self.setup_tmp = True
+            self.build_notebooks = True
+            self.build_other = True
+            self.shard_i = 0
+            self.shard_n = 1
+        elif spec == "collect":
+            self.setup_tmp = False
+            self.build_notebooks = False
+            self.build_other = True
+            self.shard_i = 0
+            self.shard_n = 1
+        else:
+            i_str, n_str = spec.split("/")
```

Comment:
Might be friendlier to explicitly validate the input rather than rely on split and int. (As an aside, at the moment "٢/٤" is valid input.)

Reply:
> Might be friendlier to explicitly validate the input rather than rely on split and int. (As an aside, at the moment "٢/٤" is valid input.)

What would you do? Write a regular expression? I don't think that's worth it. Remember this isn't directly customer facing code - it only runs in CI.